File: backend/src/utils/load-config.ts
import type { IAppConfig } from '@interfaces/config'
import type { TMaybe } from '@interfaces/maybe'

import appConfig from '../../configs/app.local.json'

export const loadConfig = (mode: string): TMaybe<IAppConfig> => {
  switch (mode) {
    case 'production':
      // todo: implement get config via cloud
      return null
    case 'development':
      return appConfig
  }

  return null
}

File: airflow/utils/load_config.py
import os
import json

CONFIG_LOCAL_PATH = os.path.join(os.path.dirname(__file__), '../configs/app.local.json')

def load_config(mode: str):
    if mode == 'production':
        raise NotImplementedError("Production config not implemented")
    elif mode == 'development':
        with open(CONFIG_LOCAL_PATH, 'r') as f:
            return json.load(f)
    return None
